worker-message-prioritization-distribution/src/main/java/com/github/workerframework/workermessageprioritization/redistribution/shovel/ShovelDistributor.java

```diff
@@ -82,27 +117,36 @@ public void runOnce() {
                             .endsWith(queueConsumptionTarget.getKey().getName()))) {
                         LOGGER.info("Shovel {} already exists, ignoring.", queueConsumptionTarget.getKey().getName());
                     } else {
+                        // Delete this shovel after all the messages in the staging queue have been consumed OR we have reached the
+                        // maximum amount of messages we are allowed to consume from the staging queue (whichever is lower).
+                        // This ensures the shovel is always deleted.
+                        final long numMessagesInStagingQueue = queueConsumptionTarget.getKey().getMessages();
+                        final long maxNumMessagesToConsumeFromStagingQueue = queueConsumptionTarget.getValue();
+                        final long srcDeleteAfter = Math.min(numMessagesInStagingQueue, maxNumMessagesToConsumeFromStagingQueue);
```

Comment:
This is good but I can't help think it should be handled by the ConsumptionTargetCalculator. This is fine for now, just wanted to mention it.